lnrpc/invoicerpc/invoice.proto

```diff
@@ -0,0 +1,443 @@
+syntax = "proto3";
+
+package invoicerpc;
+
+
+/**
+InvoiceService has the suffix Service to prevent name clash with the Invoice
+message.
+*/
+service InvoiceService {
+
+    /** lncli: `addinvoice`
+    AddInvoice attempts to add a new invoice to the invoice database. Any
+    duplicated invoices are rejected, therefore all invoices *must* have a
+    unique payment preimage.
+    */
+    rpc AddInvoice (AddInvoiceRequest) returns (AddInvoiceResponse) {}
+
+    /** lncli: `listinvoices`
+    ListInvoices returns a list of all the invoices currently stored within the
+    database. Any active debug invoices are ignored. It has full support for
+    paginated responses, allowing users to query for specific invoices through
+    their invoice_number. This can be done by using either the first_index_offset or
+    last_index_offset fields included in the response as the index_offset of the
+    next request. The reversed flag is set by default in order to paginate
+    backwards. If you wish to paginate forwards, you must explicitly set the
+    flag to false. If none of the parameters are specified, then the last 100
+    invoices will be returned.
+    */
+    rpc ListInvoices (ListInvoiceRequest) returns (ListInvoiceResponse) {}
```

Comment:
For my use, I don't need to list invoices at all. Keep track of the hashes myself.

Reply:
This is in the proto to have a complete proposal for a new invoice interface. Once that stabilizes it should be able to replace what we currently have in the main rpc. I am leaving the call in to prevent us coming up with a system that doesn't take it into account.